pkg/operator/status.go

```diff
@@ -222,31 +222,24 @@ func isMachineConfigPoolConfigurationValid(pool *mcfgv1.MachineConfigPool, versi
 	if len(pool.Status.Configuration.Source) == 0 {
 		return fmt.Errorf("list of MachineConfigs that were used to generate configuration for pool %s is empty", pool.GetName())
 	}
-
-	type configValidationTask struct {
-		name                 string
-		versionCheckRequired bool
-	}
-	// we check that all the machineconfigs (generated, and those that were used to create generated) were generated by correct version of the controller.
-	tasks := []configValidationTask{{
-		name:                 pool.Status.Configuration.Name,
-		versionCheckRequired: true,
-	}}
-	for _, ref := range pool.Status.Configuration.Source {
-		tasks = append(tasks, configValidationTask{name: ref.Name, versionCheckRequired: false})
+	mcs := []string{pool.Status.Configuration.Name}
+	for _, fragment := range pool.Status.Configuration.Source {
+		mcs = append(mcs, fragment.Name)
 	}
-	for _, t := range tasks {
-		mc, err := machineConfigGetter(t.name)
+	for _, mcName := range mcs {
+		mc, err := machineConfigGetter(mcName)
 		if err != nil {
 			return err
 		}
 
+		// we check that all the machineconfigs (generated, and those that were used to create generated)
+		// were generated by correct version of the controller.
 		v, ok := mc.Annotations[ctrlcommon.GeneratedByControllerVersionAnnotationKey]
-		if t.versionCheckRequired && !ok {
-			return fmt.Errorf("%s must be created by controller version %s", t.name, version)
+		if !ok {
+			return fmt.Errorf("%s must be created by controller version %s", mcName, version)
 		}
 		if ok && v != version {
```

Comment:
`t.versionCheckRequired` was missing here, effectively doing the version check for everything as long as the annotation was there, so it's just irrelevant to have that field around...

Reply:
we set the annotation anyway for _anything_ so there's little to check if the annotation is there